### PR TITLE
feat(validation): Reject unexpected app JWKs

### PR DESCRIPTION
**Pull request by rfk**
@vladikoff r?

### CODE REVIEW

File: test/deriver/DeriverUtils.js [hunk 1]
```diff
@@ -4,23 +4,89 @@
...
+        (err) => { assert.equal(err.message, 'appJwk is not on curve P-256') },
+      );
+    });
+
+    it('rejects public keys whose points are not on the curve', () => {
+      const appPublicKeyJwk = b64urlencode(JSON.stringify({
+        'kty': 'EC',
+        'crv': 'P-256',
+        'x': 'SiBn6uebjigmQqw4TpNzs3AUyCae1_sG2b9Fzhq3Fyo',
+        //'y': 'q99Xq1RWNTFpk99pdQOSjUvwELss51PkmAGCXhLfMV4'
```
Comment (vladikoff) [inline test/deriver/DeriverUtils.js]: @rfk do we need to remove this?

Comment (rfk) [reply test/deriver/DeriverUtils.js]: nice catch :-)

File: src/deriver/DeriverUtils.js [hunk 2]
```diff
@@ -22,6 +22,18 @@ class DeriverUtils {
...
+        // In the future we might open up to additional key types, but for now it's better to
+        // be strict in what we accept.
+        if (key.kty !== 'EC') {
+          throw new Error('appJwk is not an EC key');
+        }
+        if (key.get('crv') !== 'P-256') {
+          throw new Error('appJwk is not on curve P-256');
+        }
+        if (key.has('d', true)) {
+          throw new Error('appJwk includes the private key');
```
Comment (vladikoff) [inline src/deriver/DeriverUtils.js]: 👍

File: test/deriver/DeriverUtils.js (hunk 1 above)
Comment (rfk) [reply test/deriver/DeriverUtils.js]: updated to remove this line

**Pull request merged**